registration/api/v2/views.py

```diff
@@ -252,58 +254,59 @@ def theme_config(self, request, pk=None):
         PATCH" {"main-color": "default"} should revert to the default theme
         base color.
         """
-        application = self.get_object()
-        if not application.draft_theme_config:
-            return Response(
-                status=status.HTTP_400_BAD_REQUEST,
-                data={
-                    "non_field_errors": "V1 theme isn't set up yet."
-                }
-            )
-
-        # Sanitize inputs
-        serializer = ThemeSchemaSerializer(data=request.data)
-        if not serializer.is_valid():
-            return Response(serializer.errors, status=400)
-
-        # Merges the current aplication theme draft with the values modified by
-        # the user and performs validation.
-        # If the key is empty (""), the user reverted the field to the default
-        # value, and the key needs to be erased.
-        merged_theme = {
-            key: value for key, value in {
-                **application.draft_theme_config, **serializer.validated_data
-            }.items() if value
-        }
-
-        # To make sure the required values are always available, merge dict with
-        # default values dictionary.
-        safe_merged_theme = {
-            **DEFAULT_THEME,
-            **merged_theme
-        }
-
-        # TODO: Needs additional validation/filling up if button customization
-        # is enabled to prevent the validation schema from failing
-
-        # Perform validation, handle error if failed, and return updated
-        # theme_config if succeeds.
-        try:
-            theme_schema_validate(safe_merged_theme)
-        except JSONSchemaValidationError:
-            # TODO: improve schema error feedback. Needs to be done along with
-            # multiple fronend changes to allow individual fields feedback.
-            return Response(
-                status=status.HTTP_400_BAD_REQUEST,
-                data={
-                    "non_field_errors": "Schema validation failed."
-                }
-            )
-
-        application.draft_theme_config = safe_merged_theme
-        application.save()
-
-        return Response(status=status.HTTP_200_OK, data=application.draft_theme_config)
+        application = BetaTestApplication.objects.select_for_update().get(id=pk)
```

Comment:
@rusrushal13 Did you test this change?

I'm getting this error when using this endpoint:
```
TransactionManagementError at /api/v2/instances/openedx_config/2/theme_config/
select_for_update cannot be used outside of a transaction.
```
This doesn't happen in the tests because they use a transaction wrapper around the test.
Can you create a fix for this and test it before sending to review again?

Reply:
I tested it, and it's working now :+1: